docs/testing: Add more ways of running checkpatch (#181)
unikraft/docs: content/en/docs/contributing/testing.md

### Using `checkpatch.pl`

To support you in checking your coding style, we provide a tailored version of the Linux kernel's `checkpatch.pl` script in [`support/scripts/checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl).
You should run this from the `root` of the Unikraft repository because it contains a [`.checkpatch.conf`](https://github.com/unikraft/unikraft/blob/staging/.checkpatch.conf) file which disables some tests that we consider irrelevant for Unikraft.

> Please run this script on all commit you are about to submit.

Reviewer: ```suggestion
Please run this script on all commits you are about to submit.
```

$ ./support/scripts/checkpatch.pl ./000*
```

> You can also run the patch on one or multiple files before commiting.

Reviewer: ```suggestion
You can also run the `checkpatch` on one or multiple files before committing.
```

> To do this, you can run the following command from the root folder:

Reviewer: ```suggestion
To do this, you can run the following command from the root directory of your repository clone:
```

```console

> $ ./support/scripts/checkpatch.pl --no-tree -f <path/to/the file>

Reviewer: Please add some output example for this.

```

Some of the errors can be solved automatically, so you may add the `--fix-inplace` parameter:

```console

> $ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f <path/to/the/file>

Reviewer: Same here, add some output example, maybe a diff on the modified file.

$ ./support/scripts/checkpatch.pl ./000*
```

You can also run the `checkpatch` on one or multiple files before committing by using the `-f` option.

> Furthermore, some of the errors can be automatically solved by using the `--fix-inplace` option:

Reviewer: ```suggestion
Furthermore, some errors can be automatically solved by using the `--fix-inplace` option:
```

```console
$ ./support/scripts/checkpatch.pl --no-tree -f --fix-inplace <path/to/the/file>
```

For example, if you have some changes in the `lib/vfscore/vfs.h`, you can check and fix potential coding style issue with:

```console
$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
ERROR: "foo*bar" should be "foo *bar"
#576: FILE: lib/vfscore/vfs.h:576:
+int	sys_chmod(const char*path, mode_t mode);

total: 1 errors, 0 warnings, 828 lines checked

> ...

Reviewer: ```suggestion
[...]
```

$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
total: 0 errors, 0 warnings, 828 lines checked

lib/vfscore/vfs.h has no obvious style problems and is ready for submission.
```

> Note that not all the coding style errors can be solved automatically and some of them should be addresed by hand.

Reviewer: Use `alert` shortcodes for notes. See an example [here](https://github.com/unikraft/docs/blob/main/content/en/docs/contributing/docs.md?plain=1#L77-L79).

> Note that not all the coding style errors can be solved automatically and some of them should be addresed by hand.

Reviewer: ```suggestion
Note that not all the coding style errors can be solved automatically and some of them should be addressed by hand.
```

> Note that not all the coding style errors can be solved automatically and some of them should be addresed by hand.
> 
> 

Reviewer: ```suggestion
Note that not all the coding style errors can be solved automatically and some of them should be addresed by hand.

```

$ ./support/scripts/checkpatch.pl ./000*
```

You can also run the `checkpatch` on one or multiple files before committing by using the `-f` option.
Furthermore, some errors can be automatically solved by using the `--fix-inplace` option:

```console
$ ./support/scripts/checkpatch.pl --no-tree -f --fix-inplace <path/to/the/file>
```

For example, if you have some changes in the `lib/vfscore/vfs.h`, you can check and fix potential coding style issue with:

```console
$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
ERROR: "foo*bar" should be "foo *bar"
#576: FILE: lib/vfscore/vfs.h:576:
+int	sys_chmod(const char*path, mode_t mode);

total: 1 errors, 0 warnings, 828 lines checked
[...]

$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
total: 0 errors, 0 warnings, 828 lines checked

lib/vfscore/vfs.h has no obvious style problems and is ready for submission.
```

> {{% alert theme="info" %}}
> Note that not all the coding style errors can be solved automatically and some of them should be addressed by hand.
> {{% /alert %}}

Reviewer 2: ```suggestion
{{< alert theme="info" >}}
Note that not all the coding style errors can be solved automatically and some of them should be addressed by hand.
{{< /alert >}}
```

$ ./support/scripts/checkpatch.pl ./000*
```

You can also run the `checkpatch` on one or multiple files before committing.

> To do this, you can run the following command from the `root` directory of your repository clone:

Reviewer: I would group all the information here, then give the example.
Something like:

```
You can also run the `checkpatch` on one or multiple files before committing by using the `-f` option.
Some of the errors can be solved automatically by using the `--fix-inplace` option:
```

Then give one example with both running the `checkpatch` and fixing the errors and state that you should be in the clone root directory.

Author: I think there is already stated above , so I would not mention it again.

```console
$ ./support/scripts/checkpatch.pl --no-tree -f <path/to/the/file>
```

> For example, if you have some changes in the `lib/vfscore/vfs.h`, you can check the coding style with:
> ```console
> $ ./support/scripts/checkpatch.pl --no-tree -f lib/vfscore/vfs.h
> ```

Reviewer: ```suggestion
For example, if you have some changes in the `lib/vfscore/vfs.h`, you can check the coding style with:

```console
$ ./support/scripts/checkpatch.pl --no-tree -f lib/vfscore/vfs.h
```
```

Some of the errors can be solved automatically, so you may add the `--fix-inplace` parameter:

```console
$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f <path/to/the/file>
```

For example, there are some coding-style errors (like the one below) that could be automatically solved:

```console

> $ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f <path/to/the/file>

Reviewer: ```suggestion
$ ./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
```

ERROR: "foo*bar" should be "foo *bar"
#576: FILE: lib/vfscore/vfs.h:576:
+int	sys_chmod(const char*path, mode_t mode);

total: 1 errors, 0 warnings, 828 lines checked
...

> $./support/scripts/checkpatch.pl --no-tree --fix-inplace -f <path/to/the/file>

Reviewer: ```suggestion
$./support/scripts/checkpatch.pl --no-tree --fix-inplace -f lib/vfscore/vfs.h
```
